staging/src/k8s.io/apiserver/pkg/apis/config/types.go

```diff
@@ -89,4 +89,22 @@ type KMSConfiguration struct {
 	// Timeout for gRPC calls to kms-plugin (ex. 5s). The default is 3 seconds.
 	// +optional
 	Timeout *metav1.Duration
+	// HealthzEndpoint contains information required to validate healthz of kms-plugin.
+	// +optional
+	HealthzEndpoint *HealthzEndpoint
+}
+
+// HealthzEndpoint contains information required to validate healthz of kms-plugin.
+type HealthzEndpoint struct {
+	// URL is the url on which kube-apiserver could check the health status of kms-plugin.
+	URL string
+	// Timeout defines when to give-up on sending healthz probes (defaults to 90 seconds).
```

Comment:
does that mean that if KMS health checks return failure for 90 seconds, the apiserver enters a permanently unhealthy state? what is the rationale for giving up on checking this independent of whatever timeout the thing monitoring apiserver health decides to use?

Reply:
Yes, this PR suggests not to start kube-apiserver when kms-plugin is unhealthy.
However, this could be a config option.

My experience showed that when kube-apiserver is allowed to start in this condition, it enters the state where Service Accounts are created, but are not backed-up by Secrets - this leads to a very messy state. 
Furthermore, kube-apiserver fails eventually anyway, it just takes longer to do so.
I think once we move away from the situation where Service Accounts are typically backed-up by Secrets we could re-think this decision.